dart/lib/src/sentry_tracer.dart

```diff
@@ -340,11 +338,15 @@ class SentryTracer extends ISentrySpan {
 
   @override
   void setMeasurement(String name, num value, {SentryMeasurementUnit? unit}) {
-    if (finished) {
-      return;
+    _rootSpan.setMeasurement(name, value, unit: unit);
+  }
+
+  void setMeasurementFromChild(String name, num value,
+      {SentryMeasurementUnit? unit}) {
+    // We don't want to overwrite the root span measurement, if it comes from a child.
+    if (!_rootSpan.measurements.containsKey(name)) {
+      setMeasurement(name, value, unit: unit);
     }
```

Comment:
what if we handle all the measurements directly in the sentrytracer instead of keeping them in the root span? I think that would be easier so we don't need to handle recursive calls and just do `_tracer.setMeasurement(...)` from the spans